simple_ldap_user/simple_ldap_user.module

```diff
@@ -479,10 +474,23 @@ function simple_ldap_user_sync_user_to_ldap($drupal_user) {
 function simple_ldap_user_sync_user_to_drupal($drupal_user) {
   // Load the LDAP user, force a cache reset.
   $ldap_user = SimpleLdapUser::singleton($drupal_user->name, TRUE);
-
-  // Nothing to sync.
-  if (!$ldap_user->exists || (isset($drupal_user->uid) && $drupal_user->uid == 1)) {
+  if(isset($drupal_user->uid) && $drupal_user->uid == 1) {
+    // Check if user is an admin.
     return;
+  } else {
+      // Sync up the status.
+      // Check the status between $drupal_user and $ldap_user before execute dp_update
+      $ldap_user_status = ($ldap_user->exists)? 1 : 0;
```

Comment:
The assignment of `$ldap_user_status` here will be overwritten by the next line.